app/assets/scripts/views/account.js

```diff
@@ -2,40 +2,290 @@
+      }
+    }
+  }
+
+  syncNotificationState (data) {
+    const subscriptions = _get(data, 'subscription', []);
+    if (!subscriptions.length) {
+      return;
+    }
+    let next = Object.assign({}, this.state.data);
+    subscriptions.forEach(sub => {
+      const rtype = rtypes[sub.rtype];
+      if (rtype === 'country' && sub.country) {
+        next.countries = next.countries.concat([{label: sub.country.name, value: sub.country.id.toString()}]);
+      } else if (rtype === 'region' && sub.region) {
+        next.regions = updateChecks(next.regions, sub.region.name.toString());
+      } else if (rtype === 'disasterType' && sub.dtype) {
+        next.disasterTypes = updateChecks(next.disasterTypes, sub.dtype.id.toString());
+      } else if (rtype === 'appeal' || rtype === 'event' || rtype === 'fieldReport') {
+        next[rtype] = updateChecks(next[rtype], stypes[sub.stype]);
+      } else if (rtype === 'surge') {
+        next.other = updateChecks(next.other, 'surge');
+      }
+    });
+    this.setState({ data: next });
+  }
+
+  onFieldChange (field, e) {
+    let data = _cloneDeep(this.state.data);
+    let val = e && e.target ? e.target.value : e;
+    _set(data, field, val === '' || val === null ? undefined : val);
+    this.setState({isDirty: true, data});
+  }
+
+  onSubmit (e) {
+    e.preventDefault();
+    const payload = this.serialize(this.state.data);
+    showGlobalLoading();
+    this.props._updateSubscriptions(payload);
+  }
+
+  serialize (data) {
+    let serialized = ['regions', 'disasterTypes', 'appeal', 'event', 'fieldReport']
+      .reduce((acc, currentType) => {
+        const flattened = _get(data, currentType, [])
+          .filter(d => d.checked)
+          .map(d => ({
+            type: currentType,
+            value: d.value
+          }));
+        return acc.concat(flattened);
+      }, []);
+
+    let otherNotifications = _get(data, 'other', []).filter(d => d.checked).map(d => ({
+      type: d.value,
+      value: true
+    }));
+    if (otherNotifications.length) {
+      serialized.push.apply(serialized, otherNotifications);
+    }
+
+    let countries = _get(data, 'countries', []).map(d => ({
+      type: 'countries',
+      value: d.value
+    }));
+    if (countries.length) {
+      serialized.push.apply(serialized, countries);
+    }
+
+    return serialized;
   }
 
   renderProfileAttributes (profile) {
     if (!profile.data) return null;
     const attributes = [
-      'user.username',
-      'user.email',
-      'user.first_name',
-      'user.last_name',
-      'city',
-      'org',
-      'org_type',
-      'department',
-      'position',
-      'phone_number'
+      'username',
+      'email',
+      'first_name',
+      'last_name',
+      'profile.city',
+      'profile.org',
+      'profile.org_type',
+      'profile.department',
+      'profile.position',
+      'profile.phone_number'
     ];
-    const profileData = profile.data.objects[0];
     return attributes.map(a => [
```

Comment:
Can you use `React.Fragment` instead of an array? :)
The shorter syntax is not yet supported, but with the name works fine!